native/components/message-markdown-rules.react.js

```diff
@@ -0,0 +1,155 @@
+// @flow
+
+import * as React from 'react';
+import { Text } from 'react-native';
+import SimpleMarkdown from 'simple-markdown';
+
+type State = { [string]: any };
+
+type ASTNode = SingleASTNode | Array<SingleASTNode>;
+
+type SingleASTNode = {
+  type: string,
+  [string]: any,
+};
+
+type ReactElement = React$Element<any>;
```

Comment:
We should always prefer `React.Element` types to `React$Element` types. Here's a recent comment from a Flow team member in the Flow Discord chat:

> React.Thing is better. React$Thing is the internal version of the type. The qualified types are aliases that may have default type arguments (see AbstractComponent for example)